File: src/UpdateProfile.py
# UpdateProfile.py
import tornado.web
import json
import base64

class Handler(tornado.web.RequestHandler):
    def post(self):
        J = json.loads(self.request.body)
        firstName = J["firstName"]
        lastName = J["lastName"]
        dob = J["birthDate"]
        ppic = base64.b64decode(J["pic"])
        print("WE GOT:", firstName, lastName, dob, ppic[:20])
        resp={"ok": True}
        self.write( json.dumps(resp) )

File: html/UpdateProfile.js
// UpdateProfile.js
"use strict";

function submit() {
    let file = document.getElementById("profile_pic").files[0];
    if(!file){
        console.log("No file!");
        return;
    }
    let R = new FileReader();
    R.addEventListener("load", () => {
        let profilepic = btoa(R.result);    //do base64 encoding
        let fname = document.getElementById("fname").value;
        let lname = document.getElementById("lname").value;
        let dob = document.getElementById("birthdate").value;
        let J = {
            firstName: fname,
            lastName: lname,
            birthDate: dob,
            pic: profilepic
        };
        fetch( "/updateprofile",
            {   method: "POST",
                body: JSON.stringify(J)
            }
        ).then( (resp) => {
            //can also use text(), blob(), or arrayBuffer()
            resp.json().then( (J) => {
                console.log("Server said:",J);
            });
        }).catch( (err) => {
            console.log("Uh oh",err);
        })
    });
    R.readAsBinaryString(file);
}